### DISCUSSION TITLE
mysql access denied after moving moving /nsm to RAID

### DISCUSSION THREAD

**Discussion by TechCory** [Security-Onion-Solutions/securityonion#8364] [Unsupported Versions]
I recently moved /nsm to my RAID array. RAID is internal on bare metal and working as expected according to mdadm. I let things run overnight to see what type of logs were created / indexes were updated, but I haven't seen very much telemetry since the move. I've been at this for about 6 hours now and I can't seem to resolve. While troubleshooting, so-status was showing that so-fleet and so-playbook were in an ERROR state. I've tried the following, which will get so-fleet started, but then will fall back into an ERROR state:

> systemctl disable salt-minion
> reboot
> service docker stop
> iptables -F
> systemctl enable salt-minion
> reboot

Here is the result of so-status:

Checking Docker status

    Docker ------------------------------- [ OK ]

Checking container statuses

    so-aptcacherng ----------------------- [ OK ]
    so-curator --------------------------- [ OK ]
    so-dockerregistry -------------------- [ OK ]
    so-elastalert ------------------------ [ OK ]
    so-elasticsearch --------------------- [ OK ]
    so-filebeat -------------------------- [ OK ]
    so-fleet -------------------------- [ ERROR ]
    so-grafana --------------------------- [ OK ]
    so-idstools -------------------------- [ OK ]
    so-influxdb -------------------------- [ OK ]
    so-kibana ---------------------------- [ OK ]
    so-kratos ---------------------------- [ OK ]
    so-logstash -------------------------- [ OK ]
    so-mysql ----------------------------- [ OK ]
    so-nginx ----------------------------- [ OK ]
    so-playbook ----------------------- [ ERROR ]
    so-redis ----------------------------- [ OK ]
    so-sensoroni ------------------------- [ OK ]
    so-soc ------------------------------- [ OK ]
    so-soctopus -------------------------- [ OK ]
    so-steno ----------------------------- [ OK ]
    so-strelka-backend ------------------- [ OK ]
    so-strelka-coordinator --------------- [ OK ]
    so-strelka-filestream ---------------- [ OK ]
    so-strelka-frontend ------------------ [ OK ]
    so-strelka-gatekeeper ---------------- [ OK ]
    so-strelka-manager ------------------- [ OK ]
    so-suricata -------------------------- [ OK ]
    so-telegraf -------------------------- [ OK ]
    so-wazuh ----------------------------- [ OK ]
    so-zeek ------------------------------ [ OK ]

I've tried the following, though I receive a 1045 error that root@localhost does not have access to mysql:

> sudo salt-call state.apply playbook.db_init
> sudo salt-call state.apply playbook.automation_user_create

I then ran the following, but none of the supplied passwords seem to work for the mysql root account:

> salt-call pillar.get secrets

Everything was working before I moved /nsm. I followed the SO guide for adding a disk when I moved /nsm.

This appears to be a mysql DB permissions issue. What is my next move?

Thank you

Version 2.3.140

**Comment by TechCory** [comment]
I moved /nsm back to the default location, and all is well. This seems to be a permissions issue created by /nsm being moved. Any assistance would be greatly appreciated.

**Reply by Acewiza** [reply]
I only did this once back on the prior v16, but the way it worked then was to simply rename then cp /oldnsm/* to /nsm/ - from a console with the onion down.

**Reply by TechCory** [reply]
That's what they have in the Adding a Disk documentation. That was the process I followed. I'm checking SELinux and folder permissions now

**Comment by dougburks** [comment]
Compare the original /nsm and the new /nsm and make sure the permissions and SELinux/AppArmor settings are identical.

**Reply by TechCory** [reply]
I may have found the problem. During the mv /nsm /nsm.old operation, I was receiving a message that the device or resource was busy. My recourse was to umount /nsm, but then it had nothing to copy. I believe this was my issue. Now I'm back to receiving the message that the device or resource is busy. I have followed the documentation and have been successful on getting this to work on one server, but this specific server will not give in. Here is the full output of my terminal after systemctl disable salt-minion and reboot:


root:/>sudo service docker stop
Redirecting to /bin/systemctl stop docker.service
Warning: Stopping docker.service, but it can still be activated by:
  docker.socket
root:/>sudo mv /nsm /nsm.old
mv: cannot move ‘/nsm’ to ‘/nsm.old’: Device or resource busy
root:/>so-status

Checking Docker status

    Docker ----------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]

What do you recommend?

**Reply by dougburks** [reply]
You might try using the `fuser` command to figure out what is using that directory.

**Reply by isaacgolding** [reply]
Alternately boot the system into single user mode and make your drive changes in that environment.
